File: dotnet/Selenium/BasicAuthTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;
using OpenQA.Selenium.BiDi.Network;
using System.Threading.Tasks;
using OpenQA.Selenium.BiDi;

namespace Selenium;

[TestClass]
public class BasicAuthTest : TestBase
{
    [TestMethod]
    public async Task BasicAuth()
    {
        await using var intercept = await Bidi.Network.InterceptAuthAsync(async auth =>
        {
            await auth.ContinueAsync(new AuthCredentials("admin", "admin"));
        });

        await Driver.Navigate().GoToUrlAsync("https://the-internet.herokuapp.com/basic_auth");

        Assert.AreEqual(
            "Congratulations! You must have the proper credentials.",
            Driver.FindElement(By.TagName("p")).Text);
    }
}

File: java/selenium/src/test/java/dev/selenium/BasicAuthTest.java
package dev.selenium;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.UsernameAndPassword;
import org.openqa.selenium.firefox.FirefoxDriver;

public class BasicAuthTest extends TestBase {
  @Test
  public void basicAuthTest() {
    ((FirefoxDriver) driver)
        .network()
        .addAuthenticationHandler(new UsernameAndPassword("admin", "admin"));

    //    driver.network().addAuthenticationHandler(UrlPattern.WEB, new UsernameAndPassword("admin", "admin"));

    driver.navigate().to("https://the-internet.herokuapp.com/basic_auth");

    Assertions.assertEquals(
        "Congratulations! You must have the proper credentials.",
        driver.findElement(By.tagName("p")).getText());
  }
}
